gcc/rust/parse/rust-parse-impl.h

```diff
@@ -12862,6 +12862,23 @@ Parser<ManagedTokenSource>::left_denotation (const_TokenPtr tok,
 					   std::move (outer_attrs),
 					   restrictions);
 	  }
+	else if (next_tok->get_id () == FLOAT_LITERAL)
+	  {
+	    auto current_loc = next_tok->get_locus ();
+	    auto str = next_tok->get_str ();
+	    auto dot_pos = str.find (".");
+	    auto prefix = str.substr (0, dot_pos);
+	    auto suffix = str.substr (dot_pos + 1);
+	    lexer.split_current_token (
+	      {Token::make_int (current_loc, std::move (prefix),
+				CORETYPE_PURE_DECIMAL),
+	       Token::make (DOT, current_loc + 1),
+	       Token::make_int (current_loc + 2, std::move (suffix),
+				CORETYPE_PURE_DECIMAL)});
+	    return parse_tuple_index_expr (tok, std::move (left),
+					   std::move (outer_attrs),
+					   restrictions);
+	  }
```

Comment:
can you add some documentation here and an explanation as to why this is needed?